nativelib/src/main/scala/scala/scalanative/native/package.scala

```diff
@@ -147,11 +148,14 @@ package object native {
     new String(bytes, charset)
   }
 
-  /** Convert a java.lang.String to a CString using given charset. */
+  /** Convert a java.lang.String to a CString using given charset.
+   *  The resulting c-style string is allocated using malloc, and
+   *  must be freed manually.
+   */
   def toCString(str: String,
                 charset: Charset = Charset.defaultCharset()): CString = {
     val bytes = str.getBytes(charset)
-    val cstr  = GC.malloc_atomic(bytes.length + 1).cast[Ptr[Byte]]
+    val cstr  = malloc(bytes.length + 1).cast[Ptr[Byte]]
```

Comment:
@Duhemm This is a major change that we have to do to support precise garbage collection. I've seen that you use `GC.malloc` a lot in the File I/O, those have to be updated asap to do manual malloc/free. This also includes all results of `toCString` invocations.